specs/discussions/2025-05-segments-global-history.md

```diff
@@ -0,0 +1,320 @@
+  longest-chain? Depending on the shard suffering the re-org, the impact is different. Fortunately,
+  all of them can be handled quite gracefully through the longest-chain rule.
+
+  - Upward reorgs, meaning leaf or intermediate shards that suffer a re-org, are easily detected. As
+    blocks are being submitted to the parent, the parent is able to identify that there is a
+    heaviest chain that needs to replace its current view of the child shard. This requires no
+    immediate action from the parent apart from this update of the parent's view of the heaviest
+    chain.
+  - Reorgs from intermediate shards do not involve any immediate action from its child shards. The
+    newest heaviest chain may change the way (and specific blocks) where the information about the
+    child shard is being included into the parent chain and propagated to the parent chain, but this
+    shouldn't have any additional impact.
+  - Finally, beacon chain re-orgs are the only onces that may trigger actions in the lower levels of
+    the hierarchy. When there is a heaviest chain in the beacon chain replacing the current longest
+    chain, this can trigger a re-org in some shards of the lower levels of the hierarchy that were
+    following the previous longest chain, as their beacon chain references for some of their blocks
+    may be deemed invalid. This event is also handled gracefully by the longest-chain rule. The
+    farmer entitled to create the next block in the child shard after the beacon re-org will build
+    the new block on top of the latest block in the child shard with a valid beacon chain block
+    reference after the re-org, continuing in this way constructing a valid chain in the shard.
+
+> Note: How is the beacon chain reference required to validate blocks chosen by farmers in
+> lower-level shards?
+>
+> In terms of protocol correctness, any valid beacon chain block reference can be used, as long as
+> it points to a block at a higher height and slot than the previous reference in the shard.
+> However, to avoid referencing blocks that are too recent (and might not be visible to all nodes,
+> increasing the risk of rejection), the reference implementation will use a fixed. Specifically,
+> all nodes will aim to propose the most recent valid block they see in their view of the beacon
+> chain `1 / SLOT_PROBABILITY` in the past (which is currently set to 1/6, i.e. 6 slots which is ~6
+> seconds). With this, farmers in a shard will try to set their blockchain reference to the most
+> recent valid beacon chain block. Failing to propose a block reference from the beacon chain that
+> is 6 slots behind would mean that the farmer has fell out of sync. This forces shard farmers to be
+> as up-to-date as possible with the beacon chain, while keeping as recent as possible references in
+> shard block (which will help with the linking and verification of information between shards).
```

Comment:
I would rewrite this by simply defining what a valid reference is. "aiming" is not the right terminology here. Essentially for every slot produced on lower-level shard there is one and only one valid beacon chain block that must be included. Failure to include it makes block invalid.

Reply:
There is a danger with this though: an attacker may withhold blocks for extra seconds, causing invalid blocks to be produced on lower-level shards. This probably means we should tweak the depth of included block from 6-7 to something larger. The idea being that if block production is 6 seconds on average on the beacon chain, delaying block announcement by one slot might actually not make that much of a difference (statistically speaking).

Or maybe I'm just hallucinating here, someone with a better grasp of probabilities should chime in.